deployment/ccip/changeset/cs_update_rmn_config.go

```diff
@@ -0,0 +1,369 @@
+package changeset
+
+import (
+	"fmt"
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	mcmsWrappers "github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_home"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/rmn_remote"
+)
+
+type SetRMNHomeCandidateConfig struct {
+	HomeChainSelector uint64
+	RMNStaticConfig   rmn_home.RMNHomeStaticConfig
+	RMNDynamicConfig  rmn_home.RMNHomeDynamicConfig
+	DigestToOverride  [32]byte
+}
+
+func (c SetRMNHomeCandidateConfig) Validate(state CCIPOnChainState) error {
+	err := deployment.IsValidChainSelector(c.HomeChainSelector)
+	if err != nil {
+		return err
+	}
+
+	if len(c.RMNDynamicConfig.OffchainConfig) != 0 {
+		return fmt.Errorf("RMNDynamicConfig.OffchainConfig must be empty")
+	}
+	if len(c.RMNStaticConfig.OffchainConfig) != 0 {
+		return fmt.Errorf("RMNStaticConfig.OffchainConfig must be empty")
+	}
+
+	if len(c.RMNStaticConfig.Nodes) > 256 {
+		return fmt.Errorf("RMNStaticConfig.Nodes must be less than 256")
+	}
+
+	var (
+		peerIds            = make(map[[32]byte]struct{})
+		offchainPublicKeys = make(map[[32]byte]struct{})
+	)
+
+	for _, node := range c.RMNStaticConfig.Nodes {
+		if _, exists := peerIds[node.PeerId]; exists {
+			return fmt.Errorf("peerId %x is duplicated", node.PeerId)
+		}
+		peerIds[node.PeerId] = struct{}{}
+
+		if _, exists := offchainPublicKeys[node.OffchainPublicKey]; exists {
+			return fmt.Errorf("offchainPublicKey %x is duplicated", node.OffchainPublicKey)
+		}
+		offchainPublicKeys[node.OffchainPublicKey] = struct{}{}
+	}
+	rmnHome := state.Chains[c.HomeChainSelector].RMNHome
+
+	if rmnHome == nil {
+		return fmt.Errorf("RMNHome not found for chain %d", c.HomeChainSelector)
+	}
+
+	currentDigest, err := rmnHome.GetCandidateDigest(nil)
+
+	if err != nil {
+		return fmt.Errorf("failed to get RMNHome candidate digest: %w", err)
+	}
+
+	if currentDigest != c.DigestToOverride {
+		return fmt.Errorf("current digest does not match digest to override")
```

Comment:
Might be helpful to include the different digests in the error here:

```suggestion
		return fmt.Errorf("current digest (%x) does not match digest to override (%x)", currentDigest[:], c.DigestToOverride[:])
```